### PR TITLE
Refactor template logic

### PR DESCRIPTION
**Pull request by lgarvey**
Added an addons-template-map.yml file to determine which cloudformation files should be written on the environment and service level for each addon type.

This has simplified the make-storage command and removed the hard coded conditional logic, e.g `if storage_type == "s3-plicy": .... `

There is a failing WAF test, this appears to predate this PR.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@ line-length = 119
 
 [tool.poetry]
 name = "dbt-copilot-tools"
-version = "0.1.15"
+version = "0.1.16"
```
Comment (yusufsheiqh) [inline pyproject.toml:6]: You will need to rebase `main` as you're a few versions behind!

Comment (lgarvey) [reply pyproject.toml:6]: Yep, I'll do that once the previous PR is reviewed & merged.

File: commands/schemas/storage-schema.json
[the remark this answers is not in the document]
Comment (lgarvey) [reply commands/schemas/storage-schema.json:1]: Yep, I'm thinking maybe we go all out and do the rename from storage to addons everywhere, in another PR.  Let's see what the team thinks.

File: commands/copilot_cli.py [hunk 2]
```diff
@@ -135,22 +135,24 @@ def make_storage():
...
+    with open(PACKAGE_DIR / "addons-template-map.yml") as fd:
+        addon_template_map = yaml.safe_load(fd)
+
     click.echo("\n>>> Generating storage cloudformation\n")
 
     path = Path(f"copilot/environments/addons/")
     mkdir(output_dir, path)
 
     services = []
     for storage_name, storage_config in config.items():
```
Comment (WillGibson) [inline commands/copilot_cli.py:150]: Maybe...
```suggestion
    for addon_name, addon_config in config.items():
```

File: commands/copilot_cli.py [hunk 3]
```diff
@@ -135,22 +135,24 @@ def make_storage():
     templates = setup_templates()
 
     config = _validate_and_normalise_config(PACKAGE_DIR / "default-storage.yml")
-
     project_config = _validate_and_normalise_config("storage.yml")
```
Comment (WillGibson) [inline commands/copilot_cli.py:138]: Maybe...
```suggestion
    project_config = _validate_and_normalise_config("addons.yml")
```

File: commands/copilot_cli.py [hunk 4]
```diff
@@ -135,22 +135,24 @@ def make_storage():
...
+
     click.echo("\n>>> Generating storage cloudformation\n")
 
     path = Path(f"copilot/environments/addons/")
     mkdir(output_dir, path)
 
     services = []
     for storage_name, storage_config in config.items():
+        print(f">>>>>>>>> {storage_name}")
         storage_type = storage_config.pop("type")
```
Comment (WillGibson) [inline commands/copilot_cli.py:152]: Maybe...
```suggestion
        addon_type = addon_config.pop("type")
```

**Review by WillGibson — COMMENTED** [review]
I haven't looked for them all, but is it worth renaming the rest of the "storage" things to "addon" while we're at it?

File: tests/test_bootstrap_cli.py [hunk 5]
```diff
@@ -109,20 +109,17 @@ def test_make_config(tmp_path):
...
 
-    def real_line_breaks(input: any) -> str:
-        return str(input).replace("\\n", "\n")
-
     with open(str(tmp_path / "copilot/.workspace")) as workspace:
         assert workspace.read() == "application: test-app"
 
-    with open(str(tmp_path / "copilot/environments/test/manifest.yml"), "rb") as test:
-        assert real_line_breaks(test.read()) == real_line_breaks(test_environment_manifest)
+    with open(str(tmp_path / "copilot/environments/test/manifest.yml"), "r") as test:
```
Comment (WillGibson) [inline tests/test_bootstrap_cli.py:115]: Was it the opening in binary mode that made it do the line breaks as "\n" strings?

Apparently "r" is the default, so you can drop those.

**Comment by yusufsheiqh** [comment]
> Yep, I'm thinking maybe we go all out and do the rename from storage to addons everywhere, in another PR. Let's see what the team thinks.

@WillGibson I believe @lgarvey will do the renaming of all things `storage` -> `addons`  in a subsequent PR as stated above.

File: commands/copilot_cli.py (hunk 2 above)
Comment (lgarvey) [reply commands/copilot_cli.py:150]: Yeh, good shout

**Comment by gabelton** [comment]
To fix the test, you'll just have to merge in my most recent PR, so that `create_stack` isn't being called without arguments

**Pull request merged**